### PR TITLE
[BugFix] Followers could not update statistics immediately after the statistics rebuild

### LINKED ISSUE #9840: Followers could not update statistics immediately after the statistics rebuild
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9840 #10047

## Problem Summary(Required) ：
specific reproduction conditions
1. Run the analyze statement on the follower, the command will be forwarded to the master
2. Collect statistical information on the master, and delete the local cache after the master collects successfully
3. The follower runs the query statement of the related table, because the local cache is not cleared, continue to use the local cache

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 34 (35.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|1|00.00%|[287]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|1|00.00%|[212]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|[803, 816]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|17|00.00%|[142, 143, 144, 146, 147, 148, 150, 151, 173, 174, 175, 177, 178, 179, 181, 182, 183]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|8|9|88.89%|[54]|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 46 (30.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|1|00.00%|[287]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|1|00.00%|[212]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|[803, 816]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|4|00.00%|[116, 117, 166, 167]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|23|00.00%|[143, 144, 145, 147, 148, 149, 151, 152, 174, 175, 176, 178, 179, 180, 182, 183, 184, 300, 301, 302, 303, 304, 305]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|9|10|90.00%|[55]|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|1|1|100.00%|[]|

**Comment by HangyuanLiu** [comment]
run starrocks_be_build

**Comment by wanpengfei-git** [comment]
run starrocks_be_build

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java [hunk 1]
```diff
@@ -113,6 +113,9 @@ public List<ColumnStatistic> getColumnStatistics(Table table, List<String> colum
 
     @Override
     public void expireColumnStatistics(Table table, List<String> columns) {
+        if (columns == null) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java:116]: Why columns could be null in here?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java:116]: At present, it is not possible to be NULL. I just added a check here. I will change it to Preconditions.checkNotNull.

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java:116]: The column information in basicMeta is added later, there may be EditLog playback that is not recorded, then the column may be Null, here is for compatibility consideration

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Comment by Youngwb** [comment]
Could you plz explain why we could not update statistics immediately after the statistics rebuild?

**Comment by HangyuanLiu** [comment]
> Could you plz explain why we could not update statistics immediately after the statistics rebuild?

specific reproduction conditions
1. Run the analyze statement on the follower, the command will be forwarded to the master
2. Collect statistical information on the master, and delete the local cache after the master collects successfully
3. The follower runs the query statement of the related table, because the local cache is not cleared, continue to use the local cache

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 14 / 45 (31.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|1|00.00%|[287]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|1|00.00%|[212]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|[803, 816]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|3|00.00%|[116, 117, 166]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|23|00.00%|[143, 144, 145, 147, 148, 149, 151, 152, 174, 175, 176, 178, 179, 180, 182, 183, 184, 300, 301, 302, 303, 304, 305]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|9|10|90.00%|[55]|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/OlapDeleteJob.java|1|1|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**